### PR TITLE
Add ubuntu as alternate OS

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

- [ ] Test full deployment with ubuntu
- [ ] Test kata with all pods.

### CODE REVIEW

File: clr-k8s-examples/Vagrantfile [hunk 1]
```diff
@@ -63,8 +72,10 @@ Vagrant.configure("2") do |config|
       c.vm.hostname = vm_name
       c.vm.network :private_network, ip: ip, autostart: true
       c.vm.provider :libvirt do |lv|
-        lv.loader = $loader
-        lv.cpu_mode = "host-passthrough"
+        if $os == "clearlinux"
+          lv.loader = $loader
+          lv.cpu_mode = "host-passthrough"
```
Comment (krsna1729) [inline clr-k8s-examples/Vagrantfile]: CPU mode can be passthrough for any OS

File: clr-k8s-examples/setup_system.sh [hunk 2]
```diff
@@ -6,9 +6,72 @@ set -o nounset
...
+  #Given the vagrant box we are using we need to really really do this hack for networking to work.
+  # Check https://github.com/lavabit/robox/issues/35
+  sudo sed -i "/nameservers/d" /etc/netplan/01-netcfg.yaml
+  sudo sed -i "/addresses/d" /etc/netplan/01-netcfg.yaml
+  sudo sed -i "s/DNS=.*/DNS=/g" /etc/systemd/resolved.conf
+  sudo sed -i "s/DNSSEC=.*/DNSSEC=no/g" /etc/systemd/resolved.conf
+  sudo netplan generate
+  sudo netplan apply
+  sudo systemctl restart systemd-networkd.service systemd-resolved.service
+
```
Comment (krsna1729) [inline clr-k8s-examples/setup_system.sh]: How much of the above is box specific vs server install of ubuntu 1804?
Should we split up the script per OS with maintainable duplication for sake of simplicity?

File: clr-k8s-examples/setup_system.sh [hunk 3]
```diff
@@ -6,9 +6,72 @@ set -o nounset
...
+  # Apparently doing this is messing with grub and other setting. For now go without.
+  #TODO: Figure out how new boxes work and enable this.
+  #sudo -E apt upgrade -y
+
+  sudo apt install -y apt-transport-https curl
+  curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo -E apt-key add -
+  sudo bash -c 'cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
+deb https://apt.kubernetes.io/ kubernetes-xenial main
+EOF'
+  sudo -E add-apt-repository -y ppa:projectatomic/ppa
```
Comment (krsna1729) [inline clr-k8s-examples/setup_system.sh]: did this actually work? kudos. it never worked OOTB for me

File: clr-k8s-examples/Vagrantfile (hunk 1 above)
Comment (krsna1729) [reply clr-k8s-examples/Vagrantfile]: actually it needs to be, otherwise kata should not work right?

File: clr-k8s-examples/setup_system.sh (hunk 2 above)
Comment (krsna1729) [reply clr-k8s-examples/setup_system.sh]: That way end users dont have to deal with quirks introduced by us supporting multiple vagrant environments.

Comment (krsna1729) [reply clr-k8s-examples/setup_system.sh]: Test without Vagrant and see where these scripts fails

Comment (krsna1729) [reply clr-k8s-examples/setup_system.sh]: Capture change in a README update

**Comment by krsna1729** [comment]
@mcastelino, capturing conclusion from a discussion @ganeshmaharaj @AntonioMeireles and I had on CL slack about this PR - 

Better to have Vagrant setup part standalone copy on @ganeshmaharaj's github for when somebody want to test ubuntu and any other OS for k8s installation.

This repo will have the setup only for CL and YAMLs which generally are OS-agnostic anyway. No extra testing to regularly maintain other OS.

**Comment by ganeshmaharaj** [comment]
I now have this setup at https://github.com/ganeshmaharaj/vagrant-stuff/k8s/

**Comment by krsna1729** [comment]
https://github.com/ganeshmaharaj/vagrant-stuff/tree/master/k8s

**Pull request closed without merging**